File: client/src/items/types/wheel/definition.ts
import { type ItemTypeClientDefinition } from '../shared';

/** Default client-side UI definition for wheel items. */
export const wheelDefinition: ItemTypeClientDefinition = {
  type: 'wheel',
  label: 'wheel',
  editableProperties: ['title', 'spaces'],
  globalProperties: {
    useSound: 'sounds/spin.ogg',
    emitSound: 'none',
    useCooldownMs: 4000,
    emitRange: 15,
    directional: false,
    emitSoundSpeed: 50,
    emitSoundTempo: 50,
  },
};

File: server/app/items/types/teleporter/definition.py
"""Teleporter item static metadata and defaults."""

from __future__ import annotations

LABEL = "teleporter"
TOOLTIP = "Send yourself to a configured grid destination."
EDITABLE_PROPERTIES: tuple[str, ...] = ("title", "destination")
CAPABILITIES: tuple[str, ...] = ("editable", "deletable", "usable")
USE_SOUND: str | None = None
EMIT_SOUND: str | None = "/sounds/whirr.ogg"
USE_COOLDOWN_MS = 1000
EMIT_RANGE = 5
DIRECTIONAL = False
DEFAULT_TITLE = "Teleporter"
DEFAULT_DESTINATION = "0, 0, 0"
MAX_DESTINATION_LENGTH = 100
DEFAULT_PARAMS: dict = {"destination": DEFAULT_DESTINATION}
PARAM_KEYS: tuple[str, ...] = ("destination",)

PROPERTY_METADATA: dict[str, dict[str, object]] = {
    "title": {
        "valueType": "text",
        "tooltip": "Display name spoken and shown for this item.",
        "maxLength": 80,
    },
    "destination": {
        "valueType": "text",
        "label": "Destination",
        "tooltip": "Destination coordinates as x, y, z.",
        "maxLength": MAX_DESTINATION_LENGTH,
    },
}
